src/Series.cpp

```diff
@@ -1407,9 +1425,15 @@ AdvanceStatus Series::advance(
          * opening an iteration's file by beginning a step on it.
          * So, return now.
          */
+        iteration.get().m_closed = internal::CloseStatus::ClosedInBackend;
         return AdvanceStatus::OK;
     }
 
+    if (mode == AdvanceMode::ENDSTEP)
+    {
+        flushStep(/* doFlush = */ false);
```

Comment:
Note to self: There is some PR further down the road that defines a second version of `Series::advance()`, check if that also has this change.

Reply:
The other instance doesnt need it, its purpose is to be independent of any concrete Iteration.